UI/Forms/ProgressForm.cs

```diff
@@ -46,8 +42,7 @@ public void OnProgressUpdated(object sender, ProgressUpdateEvent e)
 
             if (e.Current == e.Total)
             {
-                if (this.InvokeRequired)
-                    this.Invoke(new EventHandler<ProgressUpdateEvent>(OnProgressCompleted), new object[] { sender, e });
+                this.Invoke(new EventHandler<ProgressUpdateEvent>(OnProgressCompleted), new object[] { sender, e });
```

Comment:
if this cannot be called from a different thread context (we checked above) do:

`OnProgressCompleted?.Invoke(sender, e)`

Reply:
It's even simpler:
`OnProgressCompleted(sender, e);`

Reply:
That's not a good practice of Invoking an EventHandler - it works when you're absolutely 100% sure that a handler was registered before. I didn't check if this is the case, but let's stick to common practice.

Reply:
There is no Event which can be called in ProgressForm. There only is the method.